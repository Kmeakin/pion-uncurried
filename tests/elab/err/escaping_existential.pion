enum Any {
    MkAny(A: Type, a: A),
}

let get: fn(_:Any) -> ?_
= fn(it) => match it {
    MkAny(A, a) => a,
};