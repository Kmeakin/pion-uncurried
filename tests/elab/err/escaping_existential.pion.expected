enum Any: Type {
    MkAny(A: Type, a: A): Any,
}

let get: fn(_: Any) -> #error = fn(it: Any) => match it {
    MkAny(A, a) => #error,
};

====DIAGNOSTICS====
Error: Unable to infer expr of hole
   ╭─[tests/elab/err/escaping_existential.pion:5:23]
   │
 5 │ let get: fn(_:Any) -> ?_
   ·                       │ 
   ·                       ╰─ here
───╯
Error: Type mismatch
   ╭─[tests/elab/err/escaping_existential.pion:7:20]
   │
 7 │     MkAny(A, a) => a,
   ·                    │ 
   ·                    ╰─ Help: expected ?ty(it), got A
   ·                    │ 
   ·                    ╰─ local variable escapes solution
───╯
